boards/xtensa/intel_adsp_cavs15/tools/adsplog.py

```diff
@@ -40,31 +40,39 @@
 WIN_SIZE = 0x20000
 LOG_OFFSET = WIN_OFFSET + WIN_IDX * WIN_SIZE
 
-# List of known ADSP devices by their PCI IDs
-DEVICES = ["8086:5a98"]
-
 mem = None
+pcid="/sys/bus/pci/devices"
+
+for fname in os.listdir(pcid):
+    pfile = pcid + "/" + fname + "/class"
+    if not os.path.exists(pfile):
+        continue
+    fd = open(pfile)
+    pciclass = fd.read()
 
-for dev in DEVICES:
-    # Find me a way to do this detection as cleanly in python as shell, I
-    # dare you.
-    barfile = subprocess.Popen(["sh", "-c",
-                                "echo -n "
-                                "$(dirname "
-                                f"  $(fgrep PCI_ID={dev.upper()} "
-                                "    /sys/bus/pci/devices/*/uevent))"
-                                "/resource4"],
-                               stdout=subprocess.PIPE).stdout.read()
-    if not os.path.exists(barfile):
+    pfile = pcid + "/" + fname + "/vendor"
+    if not os.path.exists(pfile):
         continue
+    fd = open(pfile)
+    pcivendor = fd.read()
 
-    if not os.access(barfile, os.R_OK):
-        sys.stderr.write(f"ERROR: Cannot open {barfile} for reading.\n")
-        sys.exit(1)
 
-    fd = open(barfile)
-    mem = mmap.mmap(fd.fileno(), MAP_SIZE, offset=LOG_OFFSET,
+    if not "0x8086" in pcivendor:
+        continue
+
+    # Intel Multimedia audio controller
+    #   0x040100 -> DSP is present
+    #   0x040380 -> DSP is present but optional
+    if "0x040100" in pciclass or "0x040380" in pciclass:
+        barfile = pcid + "/" + fname + "/resource4"
+        if not os.access(barfile, os.R_OK):
+            sys.stderr.write(f"ERROR: Cannot open {barfile} for reading.\n")
+            sys.exit(1)
+
+        fd = open(barfile)
+        mem = mmap.mmap(fd.fileno(), MAP_SIZE, offset=LOG_OFFSET,
```

Comment:
As above the `fd` does not bring anything except having to find another variable name , you can `mmap.mmap(open(barfile).fileno(), MAP_SIZE,...`

Reply:
For this I'll object a bit. the mmap.mmap line has enough activity on a single statement to my state, so I'll keep the open() on a separate line just to keep it easier to read.